src/adapter/src/coord/statement_logging.rs

```diff
@@ -104,6 +107,27 @@ impl PreparedStatementLoggingInfo {
             _sealed: sealed::Private,
         }
     }
+
+    fn insert_omitted_at_byte_limit(s: &str, byte_limit: usize) -> String {
+        if s.len() <= byte_limit {
+            return s.to_string();
+        }
+
+        // Find the character boundary just before or at `byte_limit`
+        let cutoff = s
```

Comment:
Too bad the API for this isn't stable yet: https://doc.rust-lang.org/std/primitive.str.html#method.floor_char_boundary